### PR TITLE
feat: support cross-architecture model committees

### LINKED ISSUE #1968: [Feature Request] support cross-architecture ensembles by supporting per-model training configurations
## Motivation

DPGEN model deviation can consume multiple model files, but the training side currently accepts one `default_training_param` and copies the same architecture to all `numb_models`, changing only random seeds. This prevents a first-class DPA2/DPA3/DPA4 committee from being trained and managed by one DPGEN workflow.

## Proposed behavior

Allow one training configuration per committee member while keeping the existing single-dict form backward compatible. A possible interface is:

```json
"default_training_param": [
  {"model": {"descriptor": {"type": "dpa2"}}, "...": "..."},
  {"model": {"descriptor": {"type": "dpa3"}}, "...": "..."},
  {"model": {"descriptor": {"type": "dpa4"}}, "...": "..."}
]
```

Minimum behavior:

- the list length must equal `numb_models`;
- deep-copy and normalize each member independently, including seeds, training data, type map, restart/reuse, and loss settings;
- preserve current behavior when a single dict is supplied;
- validate committee compatibility before model deviation, including type-map order, cutoff, output class, and fparam/aparam dimensions;
- add an end-to-end DPA2 + DPA3 + DPA4 regression test.

## Current limitation

The current generator reads one `default_training_param` and mutates/copies it for every model, so users can only obtain same-architecture, different-seed committees without managing heterogeneous models externally.

### PR DESCRIPTION
**Pull request by SchrodingersCattt**
## Summary

- allow default_training_param to be a dict or one independent dict per committee member
- generate per-member DeePMD inputs without mutating the shared configuration
- validate type-map order, cutoff, output class, fparam/aparam dimensions, and PT2 lower-kind compatibility
- keep global backend/model format and TensorFlow defaults unchanged
- document DPA2/DPA3/DPA4 committee configuration

Fixes #1968

## Validation

- python -m unittest tests.generator.test_deepmd_backend tests.test_check_examples
- pre-commit run --files dpgen/generator/run.py dpgen/generator/arginfo.py tests/generator/test_deepmd_backend.py doc/run/param.rst
- python -m py_compile dpgen/generator/run.py dpgen/generator/arginfo.py tests/generator/test_deepmd_backend.py

## Summary by CodeRabbit

- **New Features**
  - Cross-architecture committees support independent training configurations for each member.
  - A single configuration can still be shared across all members.
  - Member-specific descriptors and training settings are supported while required global settings remain shared.
  - Configuration lists must include one entry per committee member.
  - Random seeds are assigned across all committee model configurations.

- **Validation**
  - Added compatibility checks for type maps, cutoffs, parameter dimensions, output formats, and PT2 export settings.

- **Documentation**
  - Added configuration guidance and a JSON example for multi-member committees.

### CODE REVIEW

**Comment by SchrodingersCattt** [comment]
Addressed the review findings in 5265f22: committee validation now applies the DPA4/DPA4C backend and LAMMPS-format rules, aggregates all branch cutoffs, normalizes omitted fitting_net.type to ener, preserves Python 3.9 compatibility, and deep-copies the legacy dict form before preparation. Added cross-architecture submission coverage. All required CI checks pass.

**Comment by SchrodingersCattt** [comment]
Addressed the latest review in 6698308: DPA3 cutoff validation now reads repflow.e_rcut, seed generation covers every model_dict branch, and the new committee helpers have NumPy-style docstrings. Added regression coverage for DPA3 cutoff mismatch and model_dict branch seeds. Local 42-test suite and all required CI checks pass.

**Comment by SchrodingersCattt** [comment]
Addressed the latest outside-diff comment in 48f8c64: electron-temperature dimensions now apply to every model_dict branch through a shared helper, with regression coverage for both fparam and aparam modes. All required CI checks pass.

**Comment by SchrodingersCattt** [comment]
Updated after merging the latest master in merge commit 39e4165. The shared-descriptor seed comment is addressed: descriptor-specific seeding is guarded without skipping fitting_net/type_embedding seeds, and a regression test covers a string descriptor reference with branch heads. The conflict is resolved and the branch is mergeable; CI is running.

**Comment by SchrodingersCattt** [comment]
Updated on latest master (d2beb89) via merge commit 39e4165. The branch conflict is resolved. The remaining shared-descriptor seed issue is fixed in the merge result: descriptor seeding is conditional, while fitting_net/type_embedding seeds are still assigned for string descriptor references; regression coverage is included. All local focused tests and current CI checks pass.

**Comment by SchrodingersCattt** [comment]
Addressed the workflow-coverage review in 9944934 and deterministic-order follow-up in 3b15f70. The regression now chains committee training task generation, model linking, and model-deviation staging for all three artifacts; model discovery is sorted for deterministic handoff. Current Python 3.10/3.12 CI, codecov, docs, pre-commit, and CodeRabbit checks pass.

**Comment by SchrodingersCattt** [comment]
Addressed the CALYPSO handoff review in f6f69a9. The DPA2/DPA3/DPA4 regression now keeps the same CALYPSO configuration through training submission, post-training artifact linking, model-deviation staging, and run_model_devi dispatch. It verifies that all three staged .pth artifacts are consumed in deterministic order and distinguishes the global type map from the model type map to cover _get_calypso_model_type_map. Local focused tests (46 + 2 example checks) and pre-commit pass; current CI is green except one Python job still running.

**Review by njzjz — APPROVED** [review]
